### PR TITLE
Fix race condition when using callAllWorkers on startup

### PR DESCRIPTION
**Pull request by mischnic**
Previously, `clearConfigCache` was called on startup and would run before `childInit` was called.

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.49s | +16.00ms |
| Cached | 473.00ms | -1.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 602.00ms | +39.00ms ⚠️ |
| dist/modern/index.110a671b.js | 1.07kb | +0.00b | 614.00ms | +39.00ms ⚠️ |
| dist/modern/index.77ce15d4.css | 77.00b | +0.00b | 892.00ms | +50.00ms ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 19.35s | -167.00ms |
| Cached | 7.29s | -123.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/PermalinkedComment.52a3fc6f.js | 4.22kb | +0.00b | 885.00ms | +140.00ms ⚠️ |
| dist/UserProfile.42c44c22.js | 1.70kb | +0.00b | 885.00ms | +140.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 479.93kb | +0.00b | 65.00ms | +16.00ms ⚠️ |
| dist/PermalinkedComment.52a3fc6f.js | 4.22kb | +0.00b | 65.00ms | +24.00ms ⚠️ |
| dist/UserProfile.42c44c22.js | 1.70kb | +0.00b | 65.00ms | +23.00ms ⚠️ |
| dist/NotFound.984005f8.js | 532.00b | +0.00b | 39.00ms | +20.00ms ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 52.00ms | +11.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.28m | +13.37s ⚠️ |
| Cached | 2.69s | +249.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.b0698ba2.js | 2.34mb | +6.00b ⚠️ | 1.52m | +13.12s ⚠️ |
| dist/editorView.4de8e34d.js | 587.39kb | +0.00b | 1.52m | +30.50s ⚠️ |
| dist/popup.c7d523d4.js | 169.35kb | +0.00b | 16.56s | +1.40s ⚠️ |
| dist/EmojiPickerComponent.13c2889c.js | 139.08kb | +0.00b | 10.38s | -983.00ms 🚀 |
| dist/Toolbar.86dd17ff.js | 99.25kb | +0.00b | 16.16s | +891.00ms ⚠️ |
| dist/media-viewer.022fdde3.js | 72.71kb | +0.00b | 11.07s | -1.25s 🚀 |
| dist/card.f3741dfa.js | 53.76kb | +0.00b | 8.66s | -536.00ms 🚀 |
| dist/card.2fa3ec33.js | 51.51kb | +0.00b | 15.90s | +3.12s ⚠️ |
| dist/component.088d2ae3.js | 30.87kb | +0.00b | 3.75s | +378.00ms ⚠️ |
| dist/esm.2811a427.js | 27.71kb | +0.00b | 10.30s | +3.17s ⚠️ |
| dist/smartMediaEditor.cfe1bdad.js | 16.79kb | +0.00b | 16.56s | +1.29s ⚠️ |
| dist/dropzone.374e8cb6.js | 16.03kb | +0.00b | 11.67s | -3.26s 🚀 |
| dist/ui.078d3b2e.js | 14.11kb | +0.00b | 8.32s | -637.00ms 🚀 |
| dist/workerHasher.83104f73.js | 11.90kb | +0.00b | 6.27s | -1.01s 🚀 |
| dist/component.1880fd07.js | 6.27kb | +0.00b | 3.97s | +378.00ms ⚠️ |
| dist/card.c2c3e64d.js | 5.71kb | +0.00b | 10.72s | +1.11s ⚠️ |
| dist/media-viewer.083ce2a1.js | 4.01kb | +0.00b | 10.59s | -924.00ms 🚀 |
| dist/EmojiPickerComponent.9a5c1b76.js | 3.68kb | +0.00b | 6.77s | -1.06s 🚀 |
| dist/index.10f02a5e.css | 3.46kb | +0.00b | 17.95s | -24.37s 🚀 |
| dist/Modal.420db102.js | 3.15kb | +0.00b | 993.00ms | +67.00ms ⚠️ |
| dist/clipboard.a67b66af.js | 2.97kb | +0.00b | 12.38s | -2.72s 🚀 |
| dist/16.93e3596b.js | 2.49kb | +0.00b | 986.00ms | +64.00ms ⚠️ |
| dist/ResourcedEmojiComponent.557284bd.js | 2.15kb | +0.00b | 8.28s | -666.00ms 🚀 |
| dist/card.09f4d8e9.js | 2.15kb | +0.00b | 11.11s | +993.00ms ⚠️ |
| dist/date.c50d5e8c.js | 1.96kb | +0.00b | 4.35s | -259.00ms 🚀 |
| dist/browser.dfefe24a.js | 1.82kb | +0.00b | 12.41s | -2.72s 🚀 |
| dist/16.aa2b1e8c.js | 1.79kb | +0.00b | 4.21s | +411.00ms ⚠️ |
| dist/workerHasher.245b987a.js | 1.75kb | +0.00b | 6.22s | -936.00ms 🚀 |
| dist/workerHasher.f5a304ed.js | 1.75kb | +0.00b | 15.92s | +3.14s ⚠️ |
| dist/list-number.41f28a70.js | 1.68kb | +0.00b | 4.49s | -276.00ms 🚀 |
| dist/code.d63c0558.js | 1.61kb | +0.00b | 4.35s | -259.00ms 🚀 |
| dist/link.a9025e77.js | 1.53kb | +0.00b | 4.49s | -276.00ms 🚀 |
| dist/heading6.053111a7.js | 1.53kb | +0.00b | 5.96s | -3.21s 🚀 |
| dist/heading3.9115aad6.js | 1.51kb | +0.00b | 4.74s | -590.00ms 🚀 |
| dist/16.b19b33ca.js | 1.51kb | +0.00b | 3.98s | -1.04s 🚀 |
| dist/16.15d9f682.js | 1.46kb | +0.00b | 4.14s | +417.00ms ⚠️ |
| dist/16.fe7e09db.js | 1.46kb | +0.00b | 3.98s | -968.00ms 🚀 |
| dist/16.7fc8af6d.js | 1.45kb | +0.00b | 987.00ms | +61.00ms ⚠️ |
| dist/heading5.731efbe4.js | 1.40kb | +0.00b | 4.74s | -1.58s 🚀 |
| dist/16.a4fa1a69.js | 1.34kb | +0.00b | 987.00ms | +65.00ms ⚠️ |
| dist/heading2.de347970.js | 1.33kb | +0.00b | 4.74s | -317.00ms 🚀 |
| dist/media-card-analytics-error-boundary.5ce95433.js | 1.30kb | +0.00b | 10.76s | +1.18s ⚠️ |
| dist/mention.1703453a.js | 1.29kb | +0.00b | 4.49s | -398.00ms 🚀 |
| dist/heading4.5089aec7.js | 1.29kb | +0.00b | 4.74s | -1.43s 🚀 |
| dist/16.55b14beb.js | 1.27kb | +0.00b | 987.00ms | +65.00ms ⚠️ |
| dist/layout.32cc11b1.js | 1.27kb | +0.00b | 4.49s | -276.00ms 🚀 |
| dist/16.45835899.js | 1.27kb | +0.00b | 4.33s | +299.00ms ⚠️ |
| dist/16.e83a0923.js | 1.26kb | +0.00b | 3.98s | -793.00ms 🚀 |
| dist/16.d315628f.js | 1.26kb | +0.00b | 3.98s | -968.00ms 🚀 |
| dist/16.71387abf.js | 1.26kb | +0.00b | 4.14s | +417.00ms ⚠️ |
| dist/divider.6cc4f21a.js | 1.25kb | +0.00b | 4.35s | -259.00ms 🚀 |
| dist/component.242a45f9.js | 1.23kb | +0.00b | 3.72s | +370.00ms ⚠️ |
| dist/16.f1c68cda.js | 1.23kb | +0.00b | 3.72s | +370.00ms ⚠️ |
| dist/decision.1f5dd6ad.js | 1.21kb | +0.00b | 4.35s | -259.00ms 🚀 |
| dist/panel-warning.7dcb372a.js | 1.21kb | +0.00b | 4.63s | -267.00ms 🚀 |
| dist/list.7d80b06a.js | 1.18kb | +0.00b | 4.49s | -281.00ms 🚀 |
| dist/heading1.c1474f93.js | 1.18kb | +0.00b | 4.74s | -305.00ms 🚀 |
| dist/panel-error.4dcbeea1.js | 1.11kb | +0.00b | 4.63s | -262.00ms 🚀 |
| dist/panel.9407e91b.js | 1.10kb | +0.00b | 4.63s | -513.00ms 🚀 |
| dist/media-picker-analytics-error-boundary.607a1e52.js | 1.05kb | +0.00b | 11.67s | +749.00ms ⚠️ |
| dist/panel-success.23208184.js | 1.05kb | +0.00b | 4.63s | -262.00ms 🚀 |
| dist/media-card-analytics-error-boundary.f92fc998.js | 1.05kb | +0.00b | 10.51s | +1.28s ⚠️ |
| dist/media-card-analytics-error-boundary.36e013e9.js | 1.05kb | +0.00b | 16.55s | +3.77s ⚠️ |
| dist/panel-note.7a985368.js | 1.05kb | +0.00b | 4.63s | -262.00ms 🚀 |
| dist/media-viewer-analytics-error-boundary.cceb3bd6.js | 995.00b | +0.00b | 10.59s | -924.00ms 🚀 |
| dist/simpleHasher.26a74474.js | 755.00b | +0.00b | 6.27s | -1.39s 🚀 |
| dist/simpleHasher.042b95c1.js | 755.00b | +0.00b | 16.56s | +3.78s ⚠️ |
| dist/index.html | 119.00b | +0.00b | 2.87s | +243.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.724f2bbc.js | 2.31mb | +0.00b | 585.00ms | +42.00ms ⚠️ |
| dist/pdfRenderer.f8527c1c.js | 1.11mb | +0.00b | 577.00ms | +48.00ms ⚠️ |
| dist/editorView.4de8e34d.js | 587.39kb | +0.00b | 600.00ms | +32.00ms ⚠️ |
| dist/popup.23306a8f.js | 169.21kb | +0.00b | 593.00ms | +55.00ms ⚠️ |
| dist/EmojiPickerComponent.13c2889c.js | 139.08kb | +0.00b | 562.00ms | +60.00ms ⚠️ |
| dist/Toolbar.7ab8152a.js | 99.13kb | +0.00b | 600.00ms | +32.00ms ⚠️ |
| dist/media-viewer.336ad151.js | 72.60kb | +0.00b | 577.00ms | +48.00ms ⚠️ |
| dist/card.f7371915.js | 53.89kb | +0.00b | 569.00ms | +55.00ms ⚠️ |
| dist/card.d41d0ab9.js | 51.65kb | +0.00b | 594.00ms | +37.00ms ⚠️ |
| dist/Modal.fcb6df16.js | 38.80kb | +0.00b | 450.00ms | +55.00ms ⚠️ |
| dist/component.088d2ae3.js | 30.87kb | +0.00b | 428.00ms | +43.00ms ⚠️ |
| dist/esm.2811a427.js | 27.71kb | +0.00b | 562.00ms | +60.00ms ⚠️ |
| dist/component.22035ce4.js | 22.40kb | +0.00b | 585.00ms | +180.00ms ⚠️ |
| dist/DatePicker.5e8e3524.js | 21.03kb | +0.00b | 537.00ms | +42.00ms ⚠️ |
| dist/smartMediaEditor.54e2469a.js | 16.69kb | +0.00b | 600.00ms | +32.00ms ⚠️ |
| dist/js.0a808d22.js | 16.55kb | +0.00b | 450.00ms | +54.00ms ⚠️ |
| dist/dropzone.fe7127ff.js | 15.96kb | +0.00b | 594.00ms | +37.00ms ⚠️ |
| dist/ui.078d3b2e.js | 14.11kb | +0.00b | 569.00ms | +55.00ms ⚠️ |
| dist/workerHasher.83104f73.js | 11.90kb | +0.00b | 562.00ms | +60.00ms ⚠️ |
| dist/component.1880fd07.js | 6.27kb | +0.00b | 464.00ms | +59.00ms ⚠️ |
| dist/card.c2c3e64d.js | 5.71kb | +0.00b | 577.00ms | +48.00ms ⚠️ |
| dist/media-viewer.083ce2a1.js | 4.01kb | +0.00b | 577.00ms | +48.00ms ⚠️ |
| dist/EmojiPickerComponent.55e1f602.js | 3.56kb | +0.00b | 569.00ms | +56.00ms ⚠️ |
| dist/png-chunks-extract.692b85d1.js | 3.55kb | +0.00b | 450.00ms | +54.00ms ⚠️ |
| dist/dropzone.c5e4485d.js | 3.44kb | +0.00b | 593.00ms | +55.00ms ⚠️ |
| dist/Modal.420db102.js | 3.15kb | +0.00b | 450.00ms | +55.00ms ⚠️ |
| dist/clipboard.a67b66af.js | 2.97kb | +0.00b | 594.00ms | +37.00ms ⚠️ |
| dist/16.93e3596b.js | 2.49kb | +0.00b | 428.00ms | +44.00ms ⚠️ |
| dist/ResourcedEmojiComponent.557284bd.js | 2.15kb | +0.00b | 569.00ms | +55.00ms ⚠️ |
| dist/card.09f4d8e9.js | 2.15kb | +0.00b | 593.00ms | +55.00ms ⚠️ |
| dist/date.c50d5e8c.js | 1.96kb | +0.00b | 493.00ms | +46.00ms ⚠️ |
| dist/images.c0aa934a.js | 1.90kb | +0.00b | 500.00ms | +50.00ms ⚠️ |
| dist/feedback.0e55f851.js | 1.86kb | +0.00b | 537.00ms | +43.00ms ⚠️ |
| dist/16.8ab9c9cf.js | 1.86kb | +0.00b | 485.00ms | +41.00ms ⚠️ |
| dist/browser.dfefe24a.js | 1.82kb | +0.00b | 594.00ms | +37.00ms ⚠️ |
| dist/16.aa2b1e8c.js | 1.79kb | +0.00b | 464.00ms | +58.00ms ⚠️ |
| dist/workerHasher.245b987a.js | 1.75kb | +0.00b | 562.00ms | +60.00ms ⚠️ |
| dist/workerHasher.f5a304ed.js | 1.75kb | +0.00b | 600.00ms | +43.00ms ⚠️ |
| dist/list-number.41f28a70.js | 1.68kb | +0.00b | 506.00ms | +52.00ms ⚠️ |
| dist/status.eb15cda1.js | 1.68kb | +0.00b | 514.00ms | +39.00ms ⚠️ |
| dist/code.d63c0558.js | 1.61kb | +0.00b | 493.00ms | +46.00ms ⚠️ |
| dist/link.a9025e77.js | 1.53kb | +0.00b | 506.00ms | +53.00ms ⚠️ |
| dist/heading6.053111a7.js | 1.53kb | +0.00b | 537.00ms | +43.00ms ⚠️ |
| dist/heading3.9115aad6.js | 1.51kb | +0.00b | 523.00ms | +47.00ms ⚠️ |
| dist/16.b19b33ca.js | 1.51kb | +0.00b | 478.00ms | +45.00ms ⚠️ |
| dist/16.15d9f682.js | 1.46kb | +0.00b | 464.00ms | +58.00ms ⚠️ |
| dist/16.fe7e09db.js | 1.46kb | +0.00b | 478.00ms | +46.00ms ⚠️ |
| dist/emoji.d5ab3e45.js | 1.45kb | +0.00b | 499.00ms | +49.00ms ⚠️ |
| dist/16.7fc8af6d.js | 1.45kb | +0.00b | 428.00ms | +43.00ms ⚠️ |
| dist/16.474c8276.js | 1.44kb | +0.00b | 492.00ms | +45.00ms ⚠️ |
| dist/16.d49a0bb7.js | 1.40kb | +0.00b | 485.00ms | +41.00ms ⚠️ |
| dist/heading5.731efbe4.js | 1.40kb | +0.00b | 537.00ms | +43.00ms ⚠️ |
| dist/expand.7a2a1b06.js | 1.38kb | +0.00b | 537.00ms | +43.00ms ⚠️ |
| dist/16.c4941d0d.js | 1.36kb | +0.00b | 485.00ms | +41.00ms ⚠️ |
| dist/16.a4fa1a69.js | 1.34kb | +0.00b | 428.00ms | +44.00ms ⚠️ |
| dist/heading2.de347970.js | 1.33kb | +0.00b | 523.00ms | +47.00ms ⚠️ |
| dist/16.ad29438e.js | 1.32kb | +0.00b | 485.00ms | +41.00ms ⚠️ |
| dist/16.3d7a389d.js | 1.31kb | +0.00b | 492.00ms | +45.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.5ce95433.js | 1.30kb | +0.00b | 577.00ms | +48.00ms ⚠️ |
| dist/mention.1703453a.js | 1.29kb | +0.00b | 507.00ms | +53.00ms ⚠️ |
| dist/heading4.5089aec7.js | 1.29kb | +0.00b | 523.00ms | -34.00ms 🚀 |
| dist/Modal.ae6f5f7a.js | 1.28kb | +0.00b | 450.00ms | +55.00ms ⚠️ |
| dist/16.55b14beb.js | 1.27kb | +0.00b | 428.00ms | +44.00ms ⚠️ |
| dist/layout.32cc11b1.js | 1.27kb | +0.00b | 500.00ms | +50.00ms ⚠️ |
| dist/16.45835899.js | 1.27kb | +0.00b | 478.00ms | +45.00ms ⚠️ |
| dist/16.e83a0923.js | 1.26kb | +0.00b | 464.00ms | -93.00ms 🚀 |
| dist/16.d315628f.js | 1.26kb | +0.00b | 478.00ms | +45.00ms ⚠️ |
| dist/16.cc8ae77e.js | 1.26kb | +0.00b | 478.00ms | +45.00ms ⚠️ |
| dist/16.71387abf.js | 1.26kb | +0.00b | 464.00ms | +58.00ms ⚠️ |
| dist/divider.6cc4f21a.js | 1.25kb | +0.00b | 499.00ms | +49.00ms ⚠️ |
| dist/quote.b905d023.js | 1.25kb | +0.00b | 514.00ms | +56.00ms ⚠️ |
| dist/component.242a45f9.js | 1.23kb | +0.00b | 585.00ms | +41.00ms ⚠️ |
| dist/action.3a733320.js | 1.23kb | +0.00b | 493.00ms | +46.00ms ⚠️ |
| dist/decision.1f5dd6ad.js | 1.21kb | +0.00b | 499.00ms | +49.00ms ⚠️ |
| dist/panel-warning.7dcb372a.js | 1.21kb | +0.00b | 513.00ms | +56.00ms ⚠️ |
| dist/16.8cd85281.js | 1.18kb | +0.00b | 485.00ms | +41.00ms ⚠️ |
| dist/list.7d80b06a.js | 1.18kb | +0.00b | 506.00ms | +52.00ms ⚠️ |
| dist/heading1.c1474f93.js | 1.18kb | +0.00b | 523.00ms | +47.00ms ⚠️ |
| dist/panel-error.4dcbeea1.js | 1.11kb | +0.00b | 507.00ms | +53.00ms ⚠️ |
| dist/panel.9407e91b.js | 1.10kb | +0.00b | 514.00ms | +57.00ms ⚠️ |
| dist/table.547bc9c1.js | 1.09kb | +0.00b | 522.00ms | +46.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.607a1e52.js | 1.05kb | +0.00b | 593.00ms | +55.00ms ⚠️ |
| dist/panel-success.23208184.js | 1.05kb | +0.00b | 513.00ms | +56.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.f92fc998.js | 1.05kb | +0.00b | 569.00ms | +55.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.36e013e9.js | 1.05kb | +0.00b | 594.00ms | +37.00ms ⚠️ |
| dist/panel-note.7a985368.js | 1.05kb | +0.00b | 585.00ms | +128.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.cceb3bd6.js | 995.00b | +0.00b | 593.00ms | +56.00ms ⚠️ |
| dist/simpleHasher.26a74474.js | 755.00b | +0.00b | 562.00ms | +60.00ms ⚠️ |
| dist/simpleHasher.042b95c1.js | 755.00b | +0.00b | 600.00ms | +43.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 503.00ms | +27.00ms ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 14.75s | -446.00ms |
| Cached | 518.00ms | -17.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Three.js | 580.68kb | +0.00b | 79.00ms | -15.00ms 🚀 |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/287164532700217868)**

**Comment by wbinnssmith** [comment]
Related: #5440

**Comment by wbinnssmith** [comment]
@mischnic out of curiosity, how did this manifest?

**Comment by mischnic** [comment]
Bildschirmfoto 2021-01-08 um 19 49 04

because `this.module` was undefined since `childInit` wasn't called

File: packages/core/workers/src/WorkerFarm.js [hunk 1]
```diff
@@ -502,15 +502,21 @@ export default class WorkerFarm extends EventEmitter {
...
-            retries: 0,
-          });
-        }),
+        (worker.ready
+          ? Promise.resolve()
+          : new Promise(res => worker.once('ready', res))
+        ).then(
+          () =>
+            new Promise((resolve, reject) =>
+              worker.call({
```
Comment (wbinnssmith) [inline packages/core/workers/src/WorkerFarm.js:511]: could this waiting logic be moved into `worker.call()` itself?

Comment (mischnic) [reply packages/core/workers/src/WorkerFarm.js:511]: We'd have to make sure that this call here
https://github.com/parcel-bundler/parcel/blob/7fb0d22dee6ddc8554c1849d4e78ade23e09ba47/packages/core/workers/src/Worker.js#L90-L101

doesn't get this treatment

Comment (mischnic) [reply packages/core/workers/src/WorkerFarm.js]: Done with `skipReadyCheck` in the `.call` options.

File: packages/core/workers/src/Worker.js [hunk 2]
```diff
@@ -143,14 +145,19 @@ export default class Worker extends EventEmitter {
...
-    this.send({
-      type: 'request',
-      idx: idx,
-      child: this.id,
-      handle: call.handle,
-      method: call.method,
-      args: call.args,
-    });
+    (this.ready || call.skipReadyCheck === true
+      ? Promise.resolve()
```
Comment (devongovett) [inline packages/core/workers/src/Worker.js:149]: Maybe we can avoid the extra promise here in the common case?

Comment (mischnic) [reply packages/core/workers/src/Worker.js]: Done

**Review by devongovett — APPROVED** [review]

**Pull request merged**